Src/Support/Google.Apis.Core/Http/ConfigurableMessageHandler.cs

```diff
@@ -609,6 +637,38 @@ bool DisposeAndReturnFalse(IDisposable disposable)
             return response;
         }
 
+        private async Task CredentialInterceptAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var actualCredential = GetActualCredential(request);
+            if (actualCredential != null)
+            {
+                await actualCredential.InterceptAsync(request, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private async Task<bool> CredentialHandleResponseAsync(HandleUnsuccessfulResponseArgs args)
+        {
+            var actualCredential = GetActualCredential(args.Request);
+            if (actualCredential != null && actualCredential is IHttpUnsuccessfulResponseHandler handler)
+            {
+                return await handler.HandleResponseAsync(args).ConfigureAwait(false);
+            }
+
+            return false;
+        }
+
+        private IHttpExecuteInterceptor GetActualCredential(HttpRequestMessage request)
```

Comment:
If by "actual" you mean "the one to use for this request", we've used the term "effective" in various other places.

This is effectively just one expression though, right?

```csharp
private IHttpExecuteInterceptor GetEffectiveCredential(HttpRequestMessage request) =>
    request.Properties.TryGetValue(CredentialKey, out var cred) && cred is IHttpExecuteInterceptor callCredential)
    ? callCredential : Credential;
```